homeassistant/components/rollershutter/command_rollershutter.py

```diff
@@ -0,0 +1,147 @@
+
+    for dev_name, properties in devices.items():
+        rollershutters.append(
+            CommandRollershutter(
+                hass,
+                properties.get('name', dev_name),
+                properties.get('upcmd', 'true'),
+                properties.get('downcmd', 'true'),
+                properties.get('stopcmd', 'true'),
+                properties.get('statecmd', 'true'),
+                properties.get(CONF_VALUE_TEMPLATE, False)))
+
+    add_devices_callback(rollershutters)
+
+
+# pylint: disable=too-many-arguments, too-many-instance-attributes
+class CommandRollershutter(RollershutterDevice):
+    """ Represents a rollershutter -  can be controlled using shell cmd. """
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, hass, name, command_up, command_down, command_stop,
+                 command_state, value_template):
+
+        self._hass = hass
+        self._name = name
+        self._state = 0  # False
+        self._command_up = command_up
+        self._command_down = command_down
+        self._command_stop = command_stop
+        self._command_state = command_state
+        self._value_template = value_template
+
+    @staticmethod
+    def _rollershutter(command):
+        """ Execute the actual commands. """
+        _LOGGER.info('Running command: %s', command)
+
+        success = (subprocess.call(command, shell=True) == 0)
+
+        if not success:
+            _LOGGER.error('Command failed: %s', command)
+
+        return success
+
+    @staticmethod
+    def _query_state_value(command):
+        """ Execute state command for return value. """
+        _LOGGER.info('Running state command: %s', command)
+
+        try:
+            return_value = subprocess.check_output(command, shell=True)
+            return return_value.strip().decode('utf-8')
+        except subprocess.CalledProcessError:
+            _LOGGER.error('Command failed: %s', command)
+
+    @staticmethod
+    def _query_state_code(command):
+        """ Execute state command for return code. """
+        _LOGGER.info('Running state command: %s', command)
+        return subprocess.call(command, shell=True) == 0
+
+    @property
+    def should_poll(self):
+        """ Only poll if we have statecmd. """
+        return self._command_state is not None
+#        return False
+
+    @property
+    def name(self):
+        """ The name of the rollershutter. """
+        return self._name
+
+    @property
+    def current_position(self):
+        """
+        Return current position of rollershutter.
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+#        return None
+        return self._state
+
+    def _query_state(self):
+        """ Query for state. """
+        if not self._command_state:
+            _LOGGER.error('No state command specified')
+            return
+        if self._value_template:
+            return CommandRollershutter._query_state_value(self._command_state)
```

Comment:
You can call staticmethods with self, so this should be fine (and look cleaner):
```
 return self._query_state_value(self._command_state)
```